src/training/communicator_nccl.h

```diff
@@ -4,8 +4,11 @@
 #include "3rd_party/threadpool.h"
 #include "tensors/gpu/cuda_helpers.h"
 
-#include "cuda_runtime.h"
+// Generated by NCCL make files in build/nccl/include
```

Comment:
Maybe change to
// we explicitly include nccl.h here first to ensure that it uses our own version (cuda_runtime.h includes it as <nccl.h> which would not find it)
(or whatever the correct explanation is)